action.yaml

```diff
@@ -0,0 +1,51 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SkyWalking Infra E2E
+description: End-to-End Tesing framework that help to set up, verify E2E tests.
+author: Apache SkyWalking
+inputs:
+  e2e-file:
+    description: File path of e2e file
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        mkdir -p skywalking-e2e/bin
+    - shell: bash
+      run: |
+        cd skywalking-e2e
+        git clone https://github.com/mrproliu/skywalking-infra-e2e.git
+        git -C skywalking-infra-e2e reset --hard 4fb6aef3f84b8d19bc38c75e4e4489d7ce1d8d71
```

Comment:
Did you find any clue about how to get the revision if a user specify it in `uses: apache/skywalking-infra-e2e@main`, for example, get the `v1.0` in `uses: apache/skywalking-infra-e2e@v1.0`?

Reply:
How SkyWalking eye works?

Reply:
GHA doesn't download the code base automatically?

Reply:
> GHA doesn't download the code base automatically?

Using the `composite` type GHA, it doesn't download the code automatically.

Reply:
There is a dilemma here, the SHA will be changed when we squash and merge, let me see